doc/developer/design/20231113_optimizer_notice_catalog.md

```diff
@@ -0,0 +1,547 @@
+# Catalog tables for Optimizer Notices
+
+- Associated:
+  - MaterializeInc/materialize#21513 (part of MaterializeInc/console#549)
+  - MaterializeInc/materialize#19850 (add `redacted_..._sql` columns to system catalog)
+
+<!--
+The goal of a design document is to thoroughly discover problems and
+examine potential solutions before moving into the delivery phase of
+a project. In order to be ready to share, a design document must address
+the questions in each of the following sections. Any additional content
+is at the discretion of the author.
+
+Note: Feel free to add or remove sections as needed. However, most design
+docs should at least keep the suggested sections.
+-->
+
+## The Problem
```

Comment:
This section looks good to me. For our current service model this also means Field Engineering can't look up prior notices and can only rely on asking for (or running) realtime EXPLAINs.